test/e2e/framework/helper/validation/certificates/certificates.go

```diff
@@ -360,7 +360,23 @@ func ExpectValidBasicConstraints(certificate *cmapi.Certificate, secret *corev1.
 		return fmt.Errorf("Expected CA basicConstraint to be %v, but got %v", certificate.Spec.IsCA, cert.IsCA)
 	}
 
-	// TODO: also validate pathLen
+	if certificate.Spec.MaxPathLen != nil {
+		expectedPathLen := int(*certificate.Spec.MaxPathLen)
+		expectPathLenZero := (expectedPathLen == 0)
+
+		if cert.MaxPathLen != expectedPathLen {
+			return fmt.Errorf("Expected pathLen basicConstraints to be %v, but got %v", expectedPathLen, cert.MaxPathLen)
+		}
+
+		if cert.MaxPathLenZero != expectPathLenZero {
+			return fmt.Errorf("Expected MaxPathLenZero on certificate to be %v, but got %v", expectPathLenZero, cert.MaxPathLenZero)
+		}
+	}
```

Comment:
Shouldn't we also check that pathLen doesn't exist when we didn't request it? i.e. use the same logic from `utl/pki` to determine what the value should be, and test it against the certificate.

Reply:
That's a slightly stronger check; it would be totally reasonable IMO for an issuer to set a pathLen by default for an CA cert that it signs.

I can add the check for sure (and certainly while we just support SelfSigned and CA it's totally valid) but it feels worth noting the difference.